framework/db/mssql/Schema.php

```diff
@@ -213,15 +223,48 @@ protected function loadColumnSchema($info)
         }
 
         $column->phpType = $this->getColumnPhpType($column);
+        $column->defaultValue = $this->getColumnDefaultValue($column, $info['column_default']);
 
-        if ($info['column_default'] == '(NULL)') {
-            $info['column_default'] = null;
-        }
-        if (!$column->isPrimaryKey && ($column->type !== 'timestamp' || $info['column_default'] !== 'CURRENT_TIMESTAMP')) {
-            $column->defaultValue = $column->phpTypecast($info['column_default']);
+        return $column;
+    }
+
+    /**
+     * Extracts column default value.
+     * @param ColumnSchema $column
+     * @param string $value default value expression
+     * @return mixed null value, a number, string or \yii\db\Expression object
+     */
+    private function getColumnDefaultValue($column, $value)
+    {
+        if ($value !== null && strlen($value) > 0 && $value{0} === '(' && strrpos($value, ')') === strlen($value) - 1) {
```

Comment:
It doesn't have to be, we're testing for a single byte character and that's always safe because utf is self synchronizing. I've [tested it for strpos](https://gist.github.com/nineinchnick/917e644df42ccd62db5c).